// components/charts/your-planet/actions.ts
import {
    ADD_CHARACTER_FOR_COMPARISON,
    DELETE_ALL_CHARACTERS_FOR_COMPARISON,
    FETCH_DATA_FOR_CHARACTERS_COMPARISON,
    FETCH_NEW_YOUR_PLANET_PAGE_REQUEST,
    FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS,
    PUT_DATA_FOR_CHARACTERS_COMPARISON,
    REMOVE_CHARACTER_FOR_COMPARISON,
    SET_DATA_YOUR_PLANET_FOR_PAGINATION,
    SET_YOUR_PLANET_CURRENT_PAGE,
    yourPlanetPaginationType
} from "components/charts/your-planet/types";


export const setYourPlanetCurrentPage = (payload: number): yourPlanetPaginationType => ({
    type: SET_YOUR_PLANET_CURRENT_PAGE,
    payload
})

export const fetchNewYourPlanetPageRequest= (payload: number): yourPlanetPaginationType => ({
    type: FETCH_NEW_YOUR_PLANET_PAGE_REQUEST,
    payload
})

export const fetchNewYourPlanetPageSuccess = (): yourPlanetPaginationType => ({
    type: FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS
})

export const setYourPlanetForPagination = (payload: any): yourPlanetPaginationType => ({
    type: SET_DATA_YOUR_PLANET_FOR_PAGINATION,
    payload
})

export const addCharacterForComparison = (payload: number): yourPlanetPaginationType => ({
    type: ADD_CHARACTER_FOR_COMPARISON,
    payload
})
export const removeCharacterForComparison = (payload: number): yourPlanetPaginationType => ({
    type: REMOVE_CHARACTER_FOR_COMPARISON,
    payload
})

export const deleteAllCharactersForComparison = (): yourPlanetPaginationType => ({
    type: DELETE_ALL_CHARACTERS_FOR_COMPARISON,
})


export const fetchCharactersDataForComparison = (payload: any): yourPlanetPaginationType => ({
    type: FETCH_DATA_FOR_CHARACTERS_COMPARISON,
    payload
})

export const putCharactersDataForComparison = (payload: any): yourPlanetPaginationType => ({
    type: PUT_DATA_FOR_CHARACTERS_COMPARISON,
    payload
})
