// components/charts/your-planet/reducer.ts
import produce from "immer";
import {
    FETCH_NEW_YOUR_PLANET_PAGE_REQUEST,
    FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS,
    SET_DATA_YOUR_PLANET_FOR_PAGINATION,
    yourPlanetPaginationType
} from "components/charts/your-planet/types";


export type YourPlanetStateType = {
    currentPage: number,
    charactersList: any,
    isLoading: boolean,
    countAllCharacters: null | number,
}

const initialState: YourPlanetStateType = {
    currentPage: 1,
    charactersList: [],
    isLoading: false,
    countAllCharacters: null as number | null
}


export const yourPlanetReducer = (state = initialState, action: yourPlanetPaginationType): YourPlanetStateType =>
    produce(state, draft => {
            switch (action.type) {

                case FETCH_NEW_YOUR_PLANET_PAGE_REQUEST:
                    draft.isLoading = true
                    draft.currentPage = action.payload
                    break

                case FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS:
                    draft.isLoading = false
                    break

                case SET_DATA_YOUR_PLANET_FOR_PAGINATION:
                    draft.isLoading = true
                    draft.charactersList.splice(0, draft.charactersList.length)
                    //@ts-ignore
                    action.payload?.results.map(p => draft.charactersList.push(p))
                    draft.countAllCharacters = action.payload?.count
                    break

                default:
                    break
            }
        }
    )
