// components/charts/your-planet/types.ts

export const SET_YOUR_PLANET_CURRENT_PAGE = 'SET_YOUR_PLANET_CURRENT_PAGE'
type setYourPlanetCurrentPageType = {
    type: typeof SET_YOUR_PLANET_CURRENT_PAGE
    payload: number
}

export const FETCH_NEW_YOUR_PLANET_PAGE_REQUEST = 'FETCH_NEW_YOUR_PLANET_PAGE_REQUEST'
type fetchYourPlanetNewPageRequestType = {
    type: typeof FETCH_NEW_YOUR_PLANET_PAGE_REQUEST
    payload: number
}

export const FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS = 'FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS'
type fetchYourPlanetNewPageSuccessType = {
    type: typeof FETCH_NEW_YOUR_PLANET_PAGE_SUCCESS
}
export const SET_DATA_YOUR_PLANET_FOR_PAGINATION = 'SET_DATA_YOUR_PLANET_FOR_PAGINATION'
export type setYourPlanetForPaginationType = {
    type: typeof SET_DATA_YOUR_PLANET_FOR_PAGINATION
    payload: any
}

export const ADD_CHARACTER_FOR_COMPARISON = 'ADD_CHARACTER_FOR_COMPARISON'
type addCharacterForComparisonType = {
    type: typeof ADD_CHARACTER_FOR_COMPARISON
    payload: number
}
export const REMOVE_CHARACTER_FOR_COMPARISON = 'REMOVE_CHARACTER_FOR_COMPARISON'
type removeCharacterForComparisonType = {
    type: typeof REMOVE_CHARACTER_FOR_COMPARISON
    payload: number
}

export const DELETE_ALL_CHARACTERS_FOR_COMPARISON = 'DELETE_ALL_CHARACTERS_FOR_COMPARISON'
export type deleteAllCharactersForComparisonType = {
    type: typeof DELETE_ALL_CHARACTERS_FOR_COMPARISON
}


export const FETCH_DATA_FOR_CHARACTERS_COMPARISON = 'FETCH_DATA_FOR_CHARACTERS_COMPARISON'
export type fetchCharactersDataForComparisonType = {
    type: typeof FETCH_DATA_FOR_CHARACTERS_COMPARISON
    payload: any
}
export const PUT_DATA_FOR_CHARACTERS_COMPARISON = 'PUT_DATA_FOR_CHARACTERS_COMPARISON'
export type putCharactersDataForComparisonType = {
    type: typeof PUT_DATA_FOR_CHARACTERS_COMPARISON,
    payload: any
}

export type yourPlanetPaginationType =
    | fetchYourPlanetNewPageRequestType
    | fetchYourPlanetNewPageSuccessType
    | setYourPlanetCurrentPageType
    | setYourPlanetForPaginationType
    | addCharacterForComparisonType
    | removeCharacterForComparisonType
    | deleteAllCharactersForComparisonType
    | fetchCharactersDataForComparisonType
    | putCharactersDataForComparisonType
