// components/charts/starships-comparison/actions.ts
import {
    ADD_SHIP_FOR_COMPARISON,
    DELETE_ALL_SHIPS_FOR_COMPARISON, FETCH_DATA_FOR_SHIPS_COMPARISON,
    FETCH_NEW_SHIPS_PAGE_REQUEST,
    FETCH_NEW_SHIPS_PAGE_SUCCESS, PUT_DATA_FOR_SHIPS_COMPARISON,
    REMOVE_SHIP_FOR_COMPARISON,
    SET_DATA_SHIPS_FOR_PAGINATION,
    SET_SHIPS_CURRENT_PAGE,
    shipsPaginationType
} from "components/charts/starships-comparison/types";



export const setShipsCurrentPage = (payload: number): shipsPaginationType => ({
    type: SET_SHIPS_CURRENT_PAGE,
    payload
})

export const fetchNewShipsPageRequest= (payload: number): shipsPaginationType => ({
    type: FETCH_NEW_SHIPS_PAGE_REQUEST,
    payload
})

export const fetchNewShipsPageSuccess = (): shipsPaginationType => ({
    type: FETCH_NEW_SHIPS_PAGE_SUCCESS
})

export const setShipsForPagination = (payload: any): shipsPaginationType => ({
    type: SET_DATA_SHIPS_FOR_PAGINATION,
    payload
})
export const addShipForComparison = (payload: number): shipsPaginationType => ({
    type: ADD_SHIP_FOR_COMPARISON,
    payload
})
export const removeShipForComparison = (payload: number): shipsPaginationType => ({
    type: REMOVE_SHIP_FOR_COMPARISON,
    payload
})

export const deleteAllShipsForComparison = (): shipsPaginationType => ({
    type: DELETE_ALL_SHIPS_FOR_COMPARISON,
})

export const fetchShipsDataForComparison = (payload: any): shipsPaginationType => ({
    type: FETCH_DATA_FOR_SHIPS_COMPARISON,
    payload
})

export const putShipsDataForComparison = (payload: any): shipsPaginationType => ({
    type: PUT_DATA_FOR_SHIPS_COMPARISON,
    payload
})
