// components/charts/starships-comparison/types.ts

export const SET_SHIPS_CURRENT_PAGE = 'SET_SHIPS_CURRENT_PAGE'
type setShipsCurrentPageType = {
    type: typeof SET_SHIPS_CURRENT_PAGE
    payload: number
}

export const FETCH_NEW_SHIPS_PAGE_REQUEST = 'FETCH_NEW_SHIPS_PAGE_REQUEST'
type fetchShipsNewPageRequestType = {
    type: typeof FETCH_NEW_SHIPS_PAGE_REQUEST
    payload: number
}

export const FETCH_NEW_SHIPS_PAGE_SUCCESS = 'FETCH_NEW_SHIPS_PAGE_SUCCESS'
type fetchShipsNewPageSuccessType = {
    type: typeof FETCH_NEW_SHIPS_PAGE_SUCCESS
}
export const SET_DATA_SHIPS_FOR_PAGINATION = 'SET_DATA_SHIPS_FOR_PAGINATION'
export type setShipsForPaginationType = {
    type: typeof SET_DATA_SHIPS_FOR_PAGINATION
    payload: any
}

export const ADD_SHIP_FOR_COMPARISON = 'ADD_SHIP_FOR_COMPARISON'
type addShipForComparisonType = {
    type: typeof ADD_SHIP_FOR_COMPARISON
    payload: number
}
export const REMOVE_SHIP_FOR_COMPARISON = 'REMOVE_SHIP_FOR_COMPARISON'
type removeShipForComparisonType = {
    type: typeof REMOVE_SHIP_FOR_COMPARISON
    payload: number
}

export const DELETE_ALL_SHIPS_FOR_COMPARISON = 'DELETE_ALL_SHIPS_FOR_COMPARISON'
export type deleteAllShipsForComparisonType = {
    type: typeof DELETE_ALL_SHIPS_FOR_COMPARISON
}

export const FETCH_DATA_FOR_SHIPS_COMPARISON = 'FETCH_DATA_FOR_SHIPS_COMPARISON'
export type fetchShipsDataForComparisonType = {
    type: typeof FETCH_DATA_FOR_SHIPS_COMPARISON
    payload: any
}
export const PUT_DATA_FOR_SHIPS_COMPARISON = 'PUT_DATA_FOR_SHIPS_COMPARISON'
export type putShipsDataForComparisonType = {
    type: typeof PUT_DATA_FOR_SHIPS_COMPARISON,
    payload: any
}


export type shipsPaginationType =
    | fetchShipsNewPageRequestType
    | fetchShipsNewPageSuccessType
    | setShipsCurrentPageType
    | setShipsForPaginationType
    | addShipForComparisonType
    | deleteAllShipsForComparisonType
    | removeShipForComparisonType
    | fetchShipsDataForComparisonType
    | putShipsDataForComparisonType
