// components/films/filmsTypes.ts
//for saga fetching
export const FETCH_ALL_FILMS_REQUEST = 'FETCH_ALL_FILMS_REQUEST';
type allFilmsRequestAction = {
    type: typeof FETCH_ALL_FILMS_REQUEST
}

export const FETCH_ALL_FILMS_SUCCESS = 'FETCH_ALL_FILMS_SUCCESS';
type allFilmsSuccessAction = {
    type: typeof FETCH_ALL_FILMS_SUCCESS
}

export const FETCH_ALL_FILMS_FAILURE= 'FETCH_ALL_FILMS_FAILURE';
type allFilmsFailureAction = {
    type: typeof FETCH_ALL_FILMS_FAILURE
}

//for reducer
export const PUT_ALL_FILMS = 'PUT_ALL_FILMS';
export type putAllFilmsActionType = {
    type: typeof PUT_ALL_FILMS,
    payload: any
}
export const PUT_FILM_FOR_CHARACTER = 'PUT_FILM_FOR_CHARACTER';
export type putFilmForCharacterActionType = {
    type: typeof PUT_FILM_FOR_CHARACTER,
    payload: any
}

export const FETCH_FILM_BY_ID_REQUESTED= 'FETCH_FILM_BY_ID_REQUESTED';
type filmByIdActionType = {
    type: typeof FETCH_FILM_BY_ID_REQUESTED
    payload: any
}




export type filmsActionTypes =
    | allFilmsRequestAction
    | allFilmsSuccessAction
    | allFilmsFailureAction
    | putAllFilmsActionType
    | filmByIdActionType
    | putFilmForCharacterActionType
