// components/films/filmsActions.ts
import {
    FETCH_ALL_FILMS_FAILURE,
    FETCH_ALL_FILMS_REQUEST,
    FETCH_ALL_FILMS_SUCCESS,
    PUT_ALL_FILMS,
    filmsActionTypes, putAllFilmsActionType, FETCH_FILM_BY_ID_REQUESTED, PUT_FILM_FOR_CHARACTER,
} from "components/films/filmsTypes";


export const allFilmsRequested = (): filmsActionTypes => {
    return {
        type: FETCH_ALL_FILMS_REQUEST
    }
}
export const allFilmsSuccess = (): filmsActionTypes => {
    return {
        type: FETCH_ALL_FILMS_SUCCESS
    }
}
export const allFilmsFailure = (): filmsActionTypes => {
    return {
        type: FETCH_ALL_FILMS_FAILURE
    }
}

export const putAllFilms = (payload: any): filmsActionTypes => {
    return {
        type: PUT_ALL_FILMS,
        payload
    }
}

export const putFilmForCharacter = (payload: any): filmsActionTypes => {
    return {
        type: PUT_FILM_FOR_CHARACTER,
        payload
    }
}

export const filmByIdRequested = (payload: any): filmsActionTypes => {
    return {
        type: FETCH_FILM_BY_ID_REQUESTED,
        payload

    }
}
