// components/quiz/quizReducer.ts
import produce from 'immer';
import {
    PUT_NEW_QUESTION,
    quizActionTypes,
    SET_NEXT_QUESTION,
    SET_RIGHT_QUESTION,
    SET_WRONG_QUESTION
} from "components/quiz/quizTypes";


export type quizStateType = {
    questionCounter: number
    question: any
    rightAnswers: number
    doneInQuestion: boolean
}

const initialState: quizStateType = {
    questionCounter: 0,
    question:{},
    rightAnswers: 0,
    doneInQuestion: false,
}

export const quizReducer = (state=initialState, action:quizActionTypes): quizStateType =>
    produce(state, draft => {
        switch(action.type){
            case SET_NEXT_QUESTION:
                draft.questionCounter++
                break;
            case SET_RIGHT_QUESTION:
                draft.rightAnswers++
                draft.doneInQuestion = true
                break;

            case SET_WRONG_QUESTION:
                draft.doneInQuestion = true
                break;

            case PUT_NEW_QUESTION:
                draft.question = action.payload
                draft.doneInQuestion = false
                break;

            default:
                break
        }
    })
