// components/quiz/questions-for-quiz/questions.ts
export const questionList = [
    {
        id:1,
        question: 'С какой фразы начинается космическая киносага Джорджа Лукаса?',
        answer1: {
            text: 'Совсем недавно на соседней планете...',
            correct: false
        },
        answer2: {
            text: 'Давным-давно в далекой-далекой галактике...',
            correct: true
        },
        answer3: {
            text: 'В тридевятом царстве, в тридесятом государстве...',
            correct: false
        },
        answer4: {
            text: 'Все счастливые семьи похожи друг на друга, каждая несчастливая семья несчастлива по-своему...',
            correct: false
        },
    },

    {
        id:2,
        question: 'Какую фамилию носит один из главных героев оригинальной трилогии «Звездные войны» по имени Хан?',
        answer1: {
            text: 'Батый',
            correct: false
        },
        answer2: {
            text: 'Циммер',
            correct: false
        },
        answer3: {
            text: 'Соло',
            correct: true
        },
        answer4: {
            text: 'Нуньен Сингх',
            correct: true
        },
    },

    {
        id:3,
        question: 'Какой космический корабль пилотирует персонаж Харрисона Форда?',
        answer1: {
            text: '«Годовалый страус»',
            correct: false
        },
        answer2: {
            text: '«Десятилетний попугай»',
            correct: false
        },
        answer3: {
            text: '«Столетняя сова»',
            correct: false
        },
        answer4: {
            text: '«Тысячелетний сокол»',
            correct: true
        },
    },

    {
        id:4,
        question: 'К какой расе принадлежит Чубакка?',
        answer1: {
            text: 'Андалы',
            correct: false
        },
        answer2: {
            text: 'Вуки',
            correct: true
        },
        answer3: {
            text: 'Клингонцы',
            correct: false
        },
        answer4: {
            text: 'Хуорны',
            correct: true
        },
    },

    {
        id:5,
        question: 'Что из перечисленного не является роботом?',
        answer1: {
            text: 'R2-D2',
            correct: false
        },
        answer2: {
            text: 'C-3PO',
            correct: false
        },
        answer3: {
            text: 'BB-8',
            correct: false
        },
        answer4: {
            text: 'АК-47',
            correct: true
        },
    },

    {
        id:6,
        question: 'Как зовут короля преступного мира планеты Татуин?',
        answer1: {
            text: 'Тони Сопрано',
            correct: false
        },
        answer2: {
            text: 'Вито Корлеоне',
            correct: false
        },
        answer3: {
            text: 'Пабло Эскобар',
            correct: false
        },
        answer4: {
            text: 'Джабба Хатт',
            correct: true
        },
    },

    {
        id:7,
        question: 'Кем был дядя Люка Скайуокера?',
        answer1: {
            text: 'Имперским штурмовиком',
            correct: false
        },
        answer2: {
            text: 'Охотником за головами',
            correct: false
        },
        answer3: {
            text: 'Фермером-влагодобытчиком',
            correct: true
        },
        answer4: {
            text: 'Мужем тети Мэй',
            correct: true
        },
    },

    {
        id:8,
        question: 'Что означает перевод японского понятия «Jidai Geki», которое Джордж Лукас использовал при создании слова «джедай»?',
        answer1: {
            text: '«Спагетти-вестерн»',
            correct: false
        },
        answer2: {
            text: '«Историко-революционный фильм»',
            correct: false
        },
        answer3: {
            text: '«Фантастическая антиутопия»',
            correct: false
        },
        answer4: {
            text: '«Историческая драма»',
            correct: true
        },
    },

    {
        id:9,
        question: 'Как полностью звучит реплика Оби-Ван Кеноби, обращенная к имперским штурмовикам?',
        answer1: {
            text: '«Это не те миллионы, что вы ищете»',
            correct: false
        },
        answer2: {
            text: '«Это не те сценаристы, что вы ищете»',
            correct: false
        },
        answer3: {
            text: '«Это не те дроиды, что вы ищете»',
            correct: true
        },
        answer4: {
            text: '«Это не те андроиды, что вы ищете»',
            correct: false
        },
    },

    {
        id:10,
        question: 'Выберите правильное написание фразы, которую используют в мире «Звездных войн» для пожелания удачи:',
        answer1: {
            text: '«Да пребудет со мной Сила»',
            correct: false
        },
        answer2: {
            text: '«Да пребудет с тобой Сила»',
            correct: true
        },
        answer3: {
            text: '«Да прибудет с тобой Сила»',
            correct: false
        },
        answer4: {
            text: '«Да прибудет с тобой Сила и гуманитарная помощь»',
            correct: false
        },
    },

    {
        id:11,
        question: 'Как называется вымышленная разумная микроскопическая форма жизни, находящаяся внутри всех живых существ, согласно вселенной «Звездных войн»?',
        answer1: {
            text: 'Бифидобактерии',
            correct: false
        },
        answer2: {
            text: 'Мидихлорианы',
            correct: true
        },
        answer3: {
            text: 'Хламидии',
            correct: false
        },
        answer4: {
            text: 'Стрептомицеты',
            correct: false
        },
    },

    {
        id:12,
        question: 'На какой планете Люк Скайуокер познакомился с магистром Йодой?',
        answer1: {
            text: 'Пандора',
            correct: false
        },
        answer2: {
            text: 'Дагоба',
            correct: true
        },
        answer3: {
            text: 'Альдебаран',
            correct: false
        },
        answer4: {
            text: 'Саракш',
            correct: false
        },
    },

    {
        id:13,
        question: 'Какой злодей из «Звездых войн» вынужден постоянно носить костюм с портативной системой жизнеобеспечения?',
        answer1: {
            text: 'Дарт Вейпер',
            correct: false
        },
        answer2: {
            text: 'Дарт Вейдер',
            correct: true
        },
        answer3: {
            text: 'Дарт Мол',
            correct: false
        },
        answer4: {
            text: 'Дарт Мега-Молл',
            correct: false
        },
    },

    {
        id:14,
        question: 'Как заканчивается самый знаменитый диалог между Дартом Вейдером и Люком Скайуокером?',
        answer1: {
            text: '«Да, я твой сын!»',
            correct: false
        },
        answer2: {
            text: '«Нет! Я твой отец!»',
            correct: true
        },
        answer3: {
            text: '«Ты уже большой, Агапит. У тебя уже начали расти волосы».',
            correct: false
        },
        answer4: {
            text: '«Думаю, нам надо чаще встречаться с другими людьми. Я позвоню тебе».',
            correct: false
        },
    },

    {
        id:15,
        question: 'Как называется самое мощное оружие Империи?',
        answer1: {
            text: '«Звезда пленительного счастья»',
            correct: false
        },
        answer2: {
            text: '«Звезда по имени Солнце»',
            correct: false
        },
        answer3: {
            text: '«Звезда жизни»',
            correct: false
        },
        answer4: {
            text: '«Звезда смерти»',
            correct: true
        },
    },

    {
        id:16,
        question: 'Кого принцесса Лея обзывает «заносчивым, скудоумным, небритым нерфопасом»?',
        answer1: {
            text: 'Губернатора Таркина',
            correct: false
        },
        answer2: {
            text: 'Джаббу Хатта',
            correct: false
        },
        answer3: {
            text: 'Хана Соло',
            correct: true
        },
        answer4: {
            text: 'Дарта Вейдера',
            correct: false
        },
    },

    {
        id:17,
        question: 'В каком веществе был заморожен Хан Соло?',
        answer1: {
            text: 'Адамантий',
            correct: false
        },
        answer2: {
            text: 'Карбонит',
            correct: true
        },
        answer3: {
            text: 'Криптонит',
            correct: false
        },
        answer4: {
            text: 'Мифрил',
            correct: false
        },
    },

    {
        id:18,
        question: 'Как называется раса похожих на медвежат существ, населяющих спутник планеты Эндор?',
        answer1: {
            text: 'Экивоки',
            correct: false
        },
        answer2: {
            text: 'Эвоки',
            correct: true
        },
        answer3: {
            text: 'Уоки-токи',
            correct: false
        },
        answer4: {
            text: 'Чмоки',
            correct: false
        },
    },

    {
        id:19,
        question: 'Сколько миллиардов долларов насчитывает состояние Джорджа Лукаса на сегодняшний день?',
        answer1: {
            text: '5,1',
            correct: true
        },
        answer2: {
            text: '666',
            correct: false
        },
        answer3: {
            text: '1138',
            correct: false
        },
        answer4: {
            text: '226×10 в восемнадцатой степени',
            correct: false
        },
    }]
