// components/charts/planets-population/hooks/usePlanetsChart.ts
import {useDispatch, useSelector} from "react-redux";
import {AppState} from "init/rootReducer";
import {PlanetsPopulationStateType} from "components/charts/planets-population/reducer";
import {useEffect, useState} from "react";


export const usePlanetsChart = () => {
    const dispatch = useDispatch();
    const {planetsForChart} = useSelector<AppState, PlanetsPopulationStateType>(state => state.planetsPopulationReducer)

    const [chartData, setChartData] = useState({})
    const [chartOptions, setChartOptions] = useState({})


    const chart = () => {
        setChartData({
            labels: ['asda', 'adsdasd', 'dffg', 'dfgdfgas22'],
            datasets: [
                {
                    label: 'Level of shit',
                    data: [12, 15, 17, 18],
                    backgroundColor: [
                        'rgba(54, 162, 235, 0.2)',
                        'rgba(255, 206, 86, 0.2)',
                        'rgba(75, 192, 192, 0.2)',
                        'rgba(153, 102, 255, 0.2)',],
                    borderColor: [
                        'rgba(255, 99, 132, 1)',
                        'rgba(54, 162, 235, 1)',
                        'rgba(255, 206, 86, 1)',
                        'rgba(75, 192, 192, 1)',
                        'rgba(153, 102, 255, 1)',

                    ],
                    borderWidth: 1,
                }
            ],
        })
    }

    const getChartOptions = () => {
        setChartOptions({
            scales: {
                yAxes: [{
                    ticks: {
                        autoSkip: true,
                        beginAtZero: true
                    }
                }],
            }

        })
    }

    return {
        planetsForChart,
        chartData,
        chart,
        chartOptions,
        getChartOptions
    }


}
