// components/charts/planets-comparison/actions.ts
import {
    ADD_PLANET_FOR_COMPARISON,
    DELETE_ALL_PLANETS_FOR_COMPARISON,
    FETCH_DATA_FOR_PLANETS_COMPARISON,
    FETCH_NEW_PLANET_PAGE_REQUEST,
    FETCH_NEW_PLANET_PAGE_SUCCESS,
    planetsPaginationType,
    PUT_DATA_FOR_PLANETS_COMPARISON,
    REMOVE_PLANET_FOR_COMPARISON,
    SET_CURRENT_PAGE,
    SET_DATA_PLANETS_FOR_PAGINATION
} from "components/charts/planets-comparison/types";


export const setCurrentPage = (payload: number): planetsPaginationType => ({
    type: SET_CURRENT_PAGE,
    payload
})

export const fetchNewPlanetsPageRequest= (payload: number): planetsPaginationType => ({
    type: FETCH_NEW_PLANET_PAGE_REQUEST,
    payload
})

export const fetchNewPlanetsPageSuccess = (): planetsPaginationType => ({
    type: FETCH_NEW_PLANET_PAGE_SUCCESS
})

export const setPlanetsForPagination = (payload: any): planetsPaginationType => ({
    type: SET_DATA_PLANETS_FOR_PAGINATION,
    payload
})
export const addPlanetForComparison = (payload: number): planetsPaginationType => ({
    type: ADD_PLANET_FOR_COMPARISON,
    payload
})
export const removePlanetForComparison = (payload: number): planetsPaginationType => ({
    type: REMOVE_PLANET_FOR_COMPARISON,
    payload
})

export const deleteAllPlanetsForComparison = (): planetsPaginationType => ({
    type: DELETE_ALL_PLANETS_FOR_COMPARISON,
})

export const fetchPlanetDataForComparison = (payload: any): planetsPaginationType => ({
    type: FETCH_DATA_FOR_PLANETS_COMPARISON,
    payload
})

export const putPlanetDataForComparison = (payload: any): planetsPaginationType => ({
    type: PUT_DATA_FOR_PLANETS_COMPARISON,
    payload
})
