// components/charts/planets-comparison/types.ts

export const SET_CURRENT_PAGE = 'SET_CURRENT_PAGE'
type setCurrentPageType = {
    type: typeof SET_CURRENT_PAGE
    payload: number
}

export const FETCH_NEW_PLANET_PAGE_REQUEST = 'FETCH_NEW_PLANET_PAGE_REQUEST'
type fetchNewPageRequestType = {
    type: typeof FETCH_NEW_PLANET_PAGE_REQUEST
    payload: number
}

export const FETCH_NEW_PLANET_PAGE_SUCCESS = 'FETCH_NEW_PLANET_PAGE_SUCCESS'
type fetchNewPageSuccessType = {
    type: typeof FETCH_NEW_PLANET_PAGE_SUCCESS
}
export const SET_DATA_PLANETS_FOR_PAGINATION = 'SET_DATA_PLANETS_FOR_PAGINATION'
export type setPlanetsForPaginationType = {
    type: typeof SET_DATA_PLANETS_FOR_PAGINATION
    payload: any
}

export const ADD_PLANET_FOR_COMPARISON = 'ADD_PLANET_FOR_COMPARISON'
type addPlanetForComparisonType = {
    type: typeof ADD_PLANET_FOR_COMPARISON
    payload: number
}
export const REMOVE_PLANET_FOR_COMPARISON = 'REMOVE_PLANET_FOR_COMPARISON'
type removePlanetForComparisonType = {
    type: typeof REMOVE_PLANET_FOR_COMPARISON
    payload: number
}

export const DELETE_ALL_PLANETS_FOR_COMPARISON = 'DELETE_ALL_PLANETS_FOR_COMPARISON'
export type deleteAllPlanetsForComparisonType = {
    type: typeof DELETE_ALL_PLANETS_FOR_COMPARISON
}

export const FETCH_DATA_FOR_PLANETS_COMPARISON = 'FETCH_DATA_FOR_PLANETS_COMPARISON'
export type fetchPlanetDataForComparisonType = {
    type: typeof FETCH_DATA_FOR_PLANETS_COMPARISON
    payload: any
}
export const PUT_DATA_FOR_PLANETS_COMPARISON = 'PUT_DATA_FOR_PLANETS_COMPARISON'
export type putPlanetDataForComparisonType = {
    type: typeof PUT_DATA_FOR_PLANETS_COMPARISON,
    payload: any
}

export type planetsPaginationType =
    | setCurrentPageType
    | fetchNewPageSuccessType
    | setPlanetsForPaginationType
    | fetchNewPageRequestType
    | addPlanetForComparisonType
    | removePlanetForComparisonType
    | deleteAllPlanetsForComparisonType
    | fetchPlanetDataForComparisonType
    | putPlanetDataForComparisonType
