// services/axios.ts
import axios from "axios";

const instance = axios.create({
    baseURL: 'https://swapi.dev/api/',
});


export const charactersApi = {
    getCharactersList() {
        return instance.get(`people/`).then(response => response.data)
    },
    getCharacter(id:number) {
        return instance.get(`people/${id}/`).then(response => response.data)
    },
}

export const filmsApi = {
    getFilmsList() {
        return instance.get(`films/`).then(response => response.data)
    },
    getFilm(id:number) {
        return instance.get(`films/${id}/`).then(response => response.data)
    },
}

export const planetsApi = {
    getPlanetsList() {
        return instance.get(`planets/`).then(response => response.data)
    },
    getPlanet(id:number) {
        return instance.get(`planets/${id}/`).then(response => response.data)
    },
}

export const starshipsApi = {
    getStarshipsList() {
        return instance.get(`starships/`).then(response => response.data)
    },
    getStarships(id:number) {
        return instance.get(`starships/${id}/`).then(response => response.data)
    },
}
