// components/charts/planets-population/reducer.ts
import produce from "immer";
import {
    FETCH_NEW_PLANET_PAGE_REQUEST,
    FETCH_NEW_PLANET_PAGE_SUCCESS,
    planetsPaginationType,
    SET_DATA_PLANETS_FOR_PAGINATION
} from "components/charts/planets-population/types";


export type PlanetsPopulationStateType = {
    currentPage: number,
    planetsList: any,
    isLoading: boolean,
    countAllPlanets: null | number,
}

const initialState: PlanetsPopulationStateType = {
    currentPage: 1,
    planetsList: [],
    isLoading: false,
    countAllPlanets: null as number | null
}


export const planetsPopulationReducer = (state = initialState, action: planetsPaginationType):PlanetsPopulationStateType =>
    produce(state, draft => {
            switch (action.type) {

                case FETCH_NEW_PLANET_PAGE_REQUEST:
                    draft.isLoading = true
                    draft.currentPage = action.payload
                    break

                case FETCH_NEW_PLANET_PAGE_SUCCESS:
                    draft.isLoading = false
                    break

                case SET_DATA_PLANETS_FOR_PAGINATION:
                    draft.isLoading = true
                    draft.planetsList.splice(0, draft.planetsList.length)
                    //@ts-ignore
                    action.payload?.results.map(p => draft.planetsList.push(p))
                    draft.countAllPlanets = action.payload?.count
                    break

                default:
                    break
            }
        }
    )
